api/task_processor/tasks.py

```diff
@@ -50,3 +50,17 @@ def clean_up_old_tasks():
                 0 : settings.TASK_DELETE_BATCH_SIZE  # noqa:E203
             ]
         ).delete()
+
+
+@register_recurring_task(
+    run_every=settings.TASK_DELETE_RUN_EVERY,
+    first_run_time=settings.TASK_DELETE_RUN_TIME,
+)
+def clean_up_old_recurring_task_runs():
+    if not settings.ENABLE_CLEAN_UP_OLD_TASKS:
+        return
+
+    now = timezone.now()
+    delete_before = now - timedelta(days=settings.RECURRING_TASK_RUN_RETENTION_DAYS)
+
+    RecurringTaskRun.objects.filter(finished_at__lt=delete_before).delete()
```

Comment:
Is there a reason that we shouldn't just use the same task as the one which removes regular tasks / task runs?

Reply:
Yeah, it's because of the name of the task mostly